src/security-council-mgmt/governors/modules/SecurityCouncilNomineeElectionGovernorCountingUpgradeable.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import "@openzeppelin/contracts-upgradeable/governance/GovernorUpgradeable.sol";
+
+// counting module for nominee elections
+abstract contract SecurityCouncilNomineeElectionGovernorCountingUpgradeable is Initializable, GovernorUpgradeable {
+    // todo: better name
+    struct NomineeElectionState {
+        mapping(address => uint256) tokensUsed;
+        mapping(address => uint256) votes;
+        address[] nominees;
+    }
+
+    // proposalId => NomineeElectionState
+    mapping(uint256 => NomineeElectionState) private _elections;
+
+    function __SecurityCouncilNomineeElectionGovernorCounting_init() internal onlyInitializing {}
+
+    function COUNTING_MODE() public pure virtual override returns (string memory) {
+        return "TODO: ???";
+    }
+
+    /// @notice returns true if the account has voted any amount for any contender in the proposal
+    function hasVoted(
+        uint256 proposalId, 
+        address account
+    ) 
+        public 
+        view 
+        override
+        returns (bool) 
+    {
+        return _elections[proposalId].tokensUsed[account] > 0;
+    }
+
+    /// @dev there is no minimum quorum for nominations, so we just return true
+    function _quorumReached(uint256) internal pure override returns (bool) {
+        return true;
+    }
+
+    /// @dev the vote always succeeds, so we just return true
+    function _voteSucceeded(uint256) internal pure override returns (bool) {
+        return true;
+    }
+
+    /// @dev This function is responsible for counting votes when they are cast.
+    ///      If this vote pushes the candidate over the line, then the candidate is added to the nominees
+    ///      and only the necessary amount of tokens will be deducted from the voter. 
+    /// @param proposalId the id of the proposal
+    /// @param account the account that is casting the vote
+    /// @param weight the amount of vote that account held at time of snapshot
+    /// @param params abi encoded (candidate, tokens) where tokens is the amount of tokens the account is using to vote
+    function _countVote(
+        uint256 proposalId,
+        address account,
+        uint8,
+        uint256 weight,
+        bytes memory params
+    ) internal virtual override {
+        // let's say params is (address candidate, uint256 tokens)
+        (address candidate, uint256 tokens) = abi.decode(params, (address, uint256));
```

Comment:
i think we should consistently used language of "votes" (vs tokens)

Reply:
agreed, that is better

Reply:
0fdc92a58ec395d27f92fc1737aa691660f6b97a